src/streamlink/plugins/huya.py

```diff
@@ -88,34 +90,64 @@ def _get_streams(self):
                             )),
                         ],
                     }],
+                    "vMultiStreamInfo": [
+                        {
+                            'sDisplayName': str,
+                            'iBitRate': int
+                        }
+                    ]
                 },
-                validate.get(("data", 0)),
                 validate.union_get(
-                    ("gameLiveInfo", "liveId"),
-                    ("gameLiveInfo", "nick"),
-                    ("gameLiveInfo", "roomName"),
-                    "gameStreamInfoList",
+                    ('data', 0, "gameLiveInfo", "liveId"),
+                    ('data', 0, "gameLiveInfo", "nick"),
+                    ('data', 0, "gameLiveInfo", "roomName"),
+                    ('data', 0, "gameStreamInfoList"),
+                    'vMultiStreamInfo'
                 ),
             ),
         ))
         if not data:
             return
 
-        self.id, self.author, self.title, streamdata = data
+        self.id, self.author, self.title, streamdata, vMultiStreamInfo = data
 
         self.session.http.headers.update({
             "Origin": "https://www.huya.com",
             "Referer": "https://www.huya.com/",
         })
 
         for cdntype, priority, streamname, flvurl, suffix, anticode in streamdata:
-            qs = {k: v for k, v in dict(parse_qsl(anticode)).items() if k in self._STREAM_URL_QUERYSTRING_PARAMS}
-            url = update_scheme("https://", f"{flvurl}/{streamname}.{suffix}")
-            url = update_qsd(url, qs)
-
-            name = f"source_{cdntype.lower()}"
-            self.QUALITY_WEIGHTS[name] = self._QUALITY_WEIGHTS_OVERRIDE.get(name, priority)
-            yield name, HTTPStream(self.session, url)
+            for q in vMultiStreamInfo:
+                iBitRate = q['iBitRate']
+                qs = {k: v for k, v in dict(parse_qsl(anticode)).items() if k in self._STREAM_URL_QUERYSTRING_PARAMS}
+                fm = qs['fm']
+                ctype = qs['ctype']
+                platform_id = 100
+                uid = random.randint(12340000, 12349999)
+                convert_uid = (uid << 8 | uid >> (32 - 8)) & 0xFFFFFFFF
+                wsTime = qs['wsTime']
+                seqid = uid + int(time.time() * 1000)
```

Comment:
Not sure if this is necessary, but you're calling `time.time()` twice for each stream URL.